### DISCUSSION TITLE
Prevent webView scaling

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13645] [Q&A] [answered]
How to disable the zoom and scroll in webView? I tired touchEnabled,  scalesPageToFit,  disableContextMenu and overScrollMode.

**Comment by m1ga** [comment, marked as answer]
I use this in my style:
```
enableZoomControls: false,
enabled: false,
touchEnabled: false
```

and you can also adjust the content of the webview e.g. with `overflow:none` or `pointerEvents:none` if you like

**Reply by beppo-ivel** [reply]
The zoom controls are disabled now but I can still scroll in the web view up and down, left and right.

**Reply by m1ga** [reply]
did you set the overflow and pointerEvents? Scrolling left/right sounds like the content isn't responsive. Depending on the content (local or remote). You can run this:
```js
web.addEventListener("load", function(){
	web.evalJS('document.querySelector("body").style.overflow = "hidden";')
})
```

It disables scolling but I can still see the "scrollbar" on a longer page. You can try:
https://stackoverflow.com/a/5274400/5193915
with this hyperloop example: https://github.com/tidev/titanium_mobile/issues/13188#issuecomment-974170802
to disable the scrollbars if that is the case for your.

If you use a local page make sure you have the correct HTML <meta> tags so your page will fit into your Webview.

**Reply by beppo-ivel** [reply]
> If you use a local page make sure you have the correct HTML tags so your page will fit into your Webview.

I disabled every customization of the webView. But if I click on it I can still pull the webView left and right and a small scrollbar appears in the button. 

view.xml:
```
...
<WebView id="d3clock" url="/d3/d3graph.html" />
...
```

d3graph.html:
```
<html>
    <head>
      <script type="text/javascript" src="d3.min.js"></script>
      <link rel="stylesheet" href="d3graph.css">
    </head>
    <body>
<!--
        <div id="myPlot"></div>
        <script type="text/javascript" src="linegraph.js"></script>
-->
   </body>
</html>

```

**Reply by m1ga** [reply]
Like I've said: set the correct HTML tags like
```html
<meta name="viewport" content="width=device-width, initial-scale=1.0">
```
and the styles I've posted above. If you'll need to test the styles inside the webview you can look at https://fromzerotoapp.com/debugging-webviews/ 
Or search some HTML tutorials about CSS styles for mobile phones

**Reply by beppo-ivel** [reply]
> Like I've said: set the correct HTML tags like

Thx, didn't read the [web view article](https://titaniumsdk.com/api/titanium/ui/webview.html) carefully.

> https://fromzerotoapp.com/debugging-webviews/

If I add  `<application android:debuggable="true"/>` to my manifest the app isn't starting in the emulator. And google-chrome removed the "Remote Devices" tab. So you have navigate to `chrome://inspect/#devices`.

**Reply by m1ga** [reply]
That's because the article is very old :wink: I've updated it with screenshots I've did right now and change the URL. It's working and starting fine in my emulator and phone. ` android:debuggable="true"` shouldn't make any issues for starting an app.
